migration/src/bin/migrate_to.rs

```diff
@@ -0,0 +1,604 @@
+use clap::{builder::PossibleValuesParser, command, Parser, ValueEnum};
+use sea_orm_migration::{
+    sea_orm::{ConnectOptions, Database, DatabaseConnection, DbErr, EntityTrait, QueryOrder},
+    seaql_migrations, MigratorTrait, SchemaManager,
+};
+use std::cmp::Ordering;
+use tracing::{error, info};
+use tracing_subscriber::EnvFilter;
+
+use migration::Migrator;
+
+#[derive(Copy, Clone, ValueEnum, Debug)]
+enum Direction {
+    Up,
+    Down,
+}
+
+/// Wraps the SeaORM migration library to provide additional features. Lets
+/// you bring migrations up or down to a particular version. Allows dry-run
+/// of migrations.
+#[derive(Parser, Debug)]
+#[command(about, version)]
+struct Args {
+    #[arg(value_enum)]
+    direction: Direction,
+    #[arg(value_parser = available_migrations())]
+    target_version: String,
+    #[arg(short, long)]
+    dry_run: bool,
+    #[arg(short = 'u', long, env = "DATABASE_URL")]
+    database_url: String,
+}
+
+#[async_std::main]
+async fn main() -> Result<(), Error> {
+    let args = Args::parse();
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    let db = Database::connect(ConnectOptions::new(args.database_url)).await?;
+    check_database_is_compatible::<Migrator>(&db).await?;
+    match args.direction {
+        Direction::Up => migrate_up::<Migrator>(&db, args.dry_run, &args.target_version).await?,
+        Direction::Down => {
+            migrate_down::<Migrator>(&db, args.dry_run, &args.target_version).await?
+        }
+    }
+    Ok(())
+}
+
+// Applied migrations must be a subset of the migrations available by the
+// MigratorTrait for this CLI to be operated safely.
+async fn check_database_is_compatible<M: MigratorTrait>(
```

Comment:
nit: we don't public rustdoc anywhere, but still, item level comments like this should be prefixed with `///`